actions/setup/action.yml

```diff
@@ -1,20 +1,35 @@
-name: Setup Go Tooling
-description: Setup Go tooling including cached Go modules.
+name: Setup Sage
+description: Setup Sage, including caching of tools and Go modules.
 
 inputs:
   cacheKey:
     description: Custom cache key used
     required: false
     default: ci
 
+  go-version:
+    description: The Go version to download (if necessary) and use. Supports semver spec and ranges.
+    required: false
+    default: ^1.17
+
+  fetch-depth:
+    description: Number of commits to fetch. 0 indicates all history for all branches and tags.
+    default: 0
+
 runs:
   using: composite
 
   steps:
+    - name: Checkout
+      uses: actions/checkout@v2.4.0
+      with:
+        # Needed for conventional commit linting.
+        fetch-depth: ${{ inputs.fetch-depth }}
+
```

Comment:
Having the checkout as the first step works because the action is defined in "another repo". Actions in the same repo cannot have the checkout as the first step, as you probably saw above :blush:

Reply:
Yup! I added a little workaround

Reply:
What if we used the workflow name as the default cache key?

Reply:
the problem is that it needs to be per job and not workflow. But the default key could obviously be the workflow name.

Then for cases where you only have a single job in a workflow, you can omit it :)

Reply:
Good point again - what about `    default: ${{ github.workflow }}.${{ github.job }}`

Reply:
if that works, 100% yes

Reply:
![image](https://user-images.githubusercontent.com/1151328/153197389-a4f3e91b-e702-4a40-a1ba-762c6bd5f654.png)

It worked beautifully